mailbox/store/src/main/java/org/apache/james/mailbox/store/search/SimpleMessageSearchIndex.java

```diff
@@ -1,178 +1,190 @@
+import org.apache.james.mailbox.model.SearchQuery.Criterion;
+import org.apache.james.mailbox.model.SearchQuery.UidCriterion;
+import org.apache.james.mailbox.model.SearchQuery.UidRange;
+import org.apache.james.mailbox.store.mail.MailboxMapper;
+import org.apache.james.mailbox.store.mail.MailboxMapperFactory;
+import org.apache.james.mailbox.store.mail.MessageMapper;
+import org.apache.james.mailbox.store.mail.MessageMapper.FetchType;
+import org.apache.james.mailbox.store.mail.MessageMapperFactory;
+import org.apache.james.mailbox.store.mail.model.FlagsFactory;
+import org.apache.james.mailbox.store.mail.model.MailboxMessage;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+/**
+ * {@link MessageSearchIndex} which just fetch {@link MailboxMessage}'s from the {@link MessageMapper} and use {@link MessageSearcher}
+ * to match them against the {@link SearchQuery}.
+ * 
+ * This works with every implementation but is SLOW.
+ * 
+ *
+ */
+public class SimpleMessageSearchIndex implements MessageSearchIndex {
+    private final MessageMapperFactory messageMapperFactory;
+    private final MailboxMapperFactory mailboxMapperFactory;
+    private final TextExtractor textExtractor;
+    private final AttachmentContentLoader attachmentContentLoader;
+
+    @Inject
+    public SimpleMessageSearchIndex(MessageMapperFactory messageMapperFactory, MailboxMapperFactory mailboxMapperFactory, TextExtractor textExtractor, AttachmentContentLoader attachmentContentLoader) {
+        this.messageMapperFactory = messageMapperFactory;
+        this.mailboxMapperFactory = mailboxMapperFactory;
+        this.textExtractor = textExtractor;
+        this.attachmentContentLoader = attachmentContentLoader;
+    }
+    
+    @Override
+    public EnumSet<SearchCapabilities> getSupportedCapabilities(EnumSet<MessageCapabilities> messageCapabilities) {
+        return EnumSet.of(SearchCapabilities.MultimailboxSearch,
+            SearchCapabilities.Text,
+            SearchCapabilities.Attachment,
+            SearchCapabilities.PartialEmailMatch,
+            SearchCapabilities.AttachmentFileName);
+    }
+    
+    /**
+     * Walks down the query tree's conjunctions to find a UidCriterion
+     * @param crits - list of Criterion to search from
+     * @return
+     *      first UidCriterion found
+     *      null - if not found
+     */
+    private static UidCriterion findConjugatedUidCriterion(List<Criterion> crits) {
+        for (Criterion crit : crits) {
+            if (crit instanceof UidCriterion) {
+                return (UidCriterion) crit;
+            } else if (crit instanceof ConjunctionCriterion) {
+                return findConjugatedUidCriterion(((ConjunctionCriterion) crit)
+                        .getCriteria());
+            }
+        }
+        return null;
```

Comment:
`Optional.empty()`

Reply:
Legacy code, nothing to do with my changes